src/duke/Assistant.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    public void addDeadline(String deadline) throws MissingInputException {
+        String[] inputs = deadline.split(SEPARATOR_SLASH, 2);
+        if(inputs.length == 1  || inputs[1].equals("")) { //no / detected
+            throw new MissingInputException();
+        }
+        addTask(new Deadline(inputs[0], inputs[1]));
+    }
+
+    public void addEvent(String event) throws MissingInputException {
+        String[] inputs = event.split(SEPARATOR_SLASH, 2);
+        if(inputs.length == 1 || inputs[1].equals("")) { //no / detected or / without input
+            throw new MissingInputException();
+        }
+        addTask(new Event(inputs[0], inputs[1]));
+    }
+
+    public void addToDo(String todo) {
+        //no MissingInputException required since ArrayOutOfBounds will be thrown in main
+        addTask(new ToDo(todo));
+    }
+
+    public void deleteTask(String input) {
+        int index = Integer.parseInt(input) - 1;
+        System.out.println("Removed task: " + tasks.get(index).getTaskName());
+        tasks.remove(index);
+    }
+    
+    public void loadFile() {
+        data = new File(FILE_NAME);
+        File directory = new File(DIRECTORY_NAME);
+        //if save file does not exist, create new file
+        //otherwise, read and load duke.data
+        try {
+            if (!directory.exists()) {
+                data.getParentFile().mkdirs();
+            }
+            if (!data.exists()) {
+                data.createNewFile();
+                System.out.println("No existing save duke.data\nNew file created");
+            }
+            else {
+                Scanner in = new Scanner(data);
+                while (in.hasNext()) {
+                    String nextLine = in.nextLine();
+                    String[] inputs = nextLine.split("\\|");
+                    switch (inputs[0]) {
+                    case "T":
+                        addToDo(inputs[2]);
+                        break;
+                    case "E":
+                        String event = inputs[2] + "/" + inputs[3];
+                        addEvent(event);
+                        break;
+                    case "D":
+                        String deadline = inputs[2] + "/" + inputs[3];
+                        addDeadline(deadline);
+                        break;
+                    default:
+                        throw new MissingInputException();
+                    }
+                    if (inputs[1].equals("[X]")) {
+                        tasks.get(tasks.size() - 1).finishTask();
+                    }
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("Error loading save file");
+        } catch (MissingInputException e) {
+            System.out.println(e + " contains invalid duke.data");
+        }
+    }
+
+    public void saveFile() {
+        try {
+            FileWriter fw = new FileWriter(FILE_NAME, false);
```

Comment:
Please consider changing the name for the FileWriter variable to "fileWriter" from "fw".